frontend/packages/kubevirt-plugin/src/k8s/requests/cdi-upload/cdi-upload-requests.ts

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable camelcase, @typescript-eslint/camelcase,no-await-in-loop */
+import { getName, getNamespace } from '@console/shared';
+import { EnhancedK8sMethods } from '../../enhancedK8sMethods/enhancedK8sMethods';
+import { DataVolumeModel, UploadTokenRequestModel } from '@console/kubevirt-plugin/src/models';
+import { V1alpha1DataVolume } from '@console/kubevirt-plugin/src/types/vm/disk/V1alpha1DataVolume';
+import { apiVersionForModel, K8sKind, K8sResourceKind } from '@console/internal/module/k8s';
+import { delay } from '../../../utils/utils';
+import { K8sCreateError } from '../../enhancedK8sMethods/errors';
+
+const PVC_STATUS_DELAY = 2 * 1000;
+
+const UPLOAD_STATES = {
+  SCHEDULED: 'UploadScheduled',
+  READY: 'UploadReady',
+};
+
+const waitForUploadReady = async (
+  dvName: string,
+  namespace: string,
+  k8sGet: (kind: K8sKind, name: string, namespace: string) => Promise<K8sResourceKind>,
+  counter: number = 20,
+) => {
+  let dv;
+  for (let i = 0; i < counter; i++) {
+    if (dv?.status?.phase === UPLOAD_STATES.READY) {
+      return true;
+    }
+    await delay(PVC_STATUS_DELAY);
+    dv = await k8sGet(DataVolumeModel, dvName, namespace);
```

Comment:
why not check the phase here once we get the dv

Reply:
I took example from your func
https://github.com/openshift/console/blob/master/frontend/packages/kubevirt-plugin/src/k8s/requests/v2v/import/import-v2vvmware.ts#L186

Reply:
the difference between these two functions is that in the former one you already have the object in the first call, here you don't